tox.ini

```diff
@@ -13,8 +13,10 @@ basepython = python3
 deps =
     py{26,27,36,37,38}: pytest-cov
     py{27,36,37,38}: pytest>=3.5.1
+    py{27,36,37,38}: -rpytest_extra_requirements.txt
     py{26,27}: mock
```

Comment:
I guess the mock dependency is there for a similar reason (the mock module is part of Python 3) to be able to mock ansible. Not sure if this should be split out, too. But also seems to add a lot of complexity to move all these possible customizations outside of tox.ini.

Reply:
Yes, I think `mock` is also an optional component for unit testing.  I suppose we could do something like this:
```
    py{3*}: -rpy3_pytest_extra_requirements.txt
    py{2*}: -rpy2_pytest_extra_requirements.txt
    py{26}: -rpy26_pytest_extra_requirements.txt
```
where the last one is renaming the current `pytest26_extra_requirements.txt` file.  (It isn't for pytest version 2.6, it is extra dependencies for pytest when using py26).

Reply:
The issue I see is that the amount of files is becoming excessive. With tox.ini there was initially one nice config file but it is becoming more and more.

For this specific problem, environment markers might help: https://www.python.org/dev/peps/pep-0508/#examples - I am not sure if they work well when also supporting Python 2.x

Reply:
ah, I see - so if it works, have a single `-rpytest_extra_requirements.txt` which has something like this:
```
# my pytesting needs mock and ansible
mock ; python_version < "3.0"
ansible ; python_version > "2.6"
ansible<2.7 ; python_version < "2.7"
idna<2.8 ; python_version < "2.7"
PyYAML<5.1; python_version < "2.7"
```
?
If so, looks like it works as expected on python 2.6 and python 3.8 (at least, the ones that travis CI uses, on ubuntu xenial).

Reply:
yes, I think this looks nicer